core/flyout_base.js

```diff
@@ -426,77 +426,58 @@ Blockly.Flyout.prototype.hide = function() {
 
 /**
  * Show and populate the flyout.
- * @param {!Array|!NodeList|string} xmlList List of blocks to show.
- *     Variables and procedures have a custom set of blocks.
+ * @param {!Array.<Blockly.utils.toolbox.ToolboxInfo>|!Array.<Node>|!NodeList|string} flyoutDef
+ *    List of contents to display in the flyout as either an array of xml an
+ *    array of Nodes, a NodeList or a string with the name of the dynamic category.
+ *    Variables and procedures have a custom set of blocks.
  */
-Blockly.Flyout.prototype.show = function(xmlList) {
+Blockly.Flyout.prototype.show = function(flyoutDef) {
   this.workspace_.setResizesEnabled(false);
   this.hide();
   this.clearOldBlocks_();
 
-  // Handle dynamic categories, represented by a name instead of a list of XML.
+  // Handle dynamic categories, represented by a name instead of a list.
   // Look up the correct category generation function and call that to get a
   // valid XML list.
-  if (typeof xmlList == 'string') {
+  if (typeof flyoutDef == 'string') {
     var fnToApply = this.workspace_.targetWorkspace.getToolboxCategoryCallback(
-        xmlList);
+        flyoutDef);
     if (typeof fnToApply != 'function') {
       throw TypeError('Couldn\'t find a callback function when opening' +
           ' a toolbox category.');
     }
-    xmlList = fnToApply(this.workspace_.targetWorkspace);
-    if (!Array.isArray(xmlList)) {
+    flyoutDef = fnToApply(this.workspace_.targetWorkspace);
+    if (!Array.isArray(flyoutDef)) {
       throw TypeError('Result of toolbox category callback must be an array.');
     }
   }
+  // contentInfo is either an array with JSON, an Array with xml, or a NodeList.
```

Comment:
This comment is confusing not because contentInfo -> flyoutDef, but because I expected it to be explaining the variable being declared here (parsedContent) instead of the one being used (flyoutDef).